reyna/src/com/b2msolutions/reyna/services/BlackoutTime.java

```diff
@@ -15,18 +15,17 @@ public boolean canSendAtTime(Calendar now, String range) throws ParseException {
 
         String[] rangesSplit = range.split(",");
         for (String rangeSplit : rangesSplit) {
-            List<Time> times = parseTime(rangeSplit);
-            TimeRange timeRange = new TimeRange(times.get(0), times.get(1));
-            if (timeRange.contains(new Time(now.get(Calendar.HOUR_OF_DAY),1))) {
+            TimeRange timeRange = parseTime(rangeSplit);
+            Time timeNow = new Time(now.get(Calendar.HOUR_OF_DAY), now.get(Calendar.MINUTE));
+            if (timeRange.contains(timeNow) && !timeRange.isEmpty()) {
```

Comment:
Now with the refactor it is very clear why this is a funny way of structuring it. How can a timeRange contain a time point and be empty at the same time. Logically that is impossible. Suggestion: can we just move the isEmpty check inside the contains function, i.e. as the first line in TimeRange.contains have "if (isEmpty()) return false;"